tensorboard/plugins/scalar/polymer3/tf_scalar_dashboard/tf-scalar-card.ts

```diff
@@ -0,0 +1,354 @@
+        color: inherit;
+        text-decoration: none;
+        white-space: nowrap;
+      }
+    </style>
+  `;
+
+  @property({type: String})
+  tag: string;
+
+  @property({type: Array})
+  dataToLoad: object[];
+
+  @property({type: String})
+  xType: string;
+
+  @property({type: Boolean})
+  active: boolean;
+
+  @property({type: Boolean})
+  ignoreYOutliers: boolean;
+
+  @property({type: Object})
+  requestManager: object;
+
+  @property({type: Boolean})
+  showDownLinks: boolean;
+
+  @property({type: Boolean})
+  smoothingEnabled: boolean;
+
+  @property({type: Number})
+  smoothingWeight: number;
+
+  @property({type: Object})
+  tagMetadata: object;
+
+  // If specified uses the given colorScale; otherwise, uses a built-in
+  // default. See _getColorScale below for more details.
+  @property({type: Object})
+  colorScale: object = null;
+
+  @property({type: String})
+  tooltipSortingMethod: string;
+
+  // This function is called when data is received from the backend.
+  @property({type: Object})
+  _loadDataCallback: object = function() {
+    return (scalarChart, datum, data) => {
+      const formattedData = data.map((datum) => ({
+        wall_time: new Date(datum[0] * 1000),
+        step: datum[1],
+        scalar: datum[2],
+      }));
+      const name = this._getSeriesNameFromDatum(datum);
+      scalarChart.setSeriesMetadata(name, datum);
+      scalarChart.setSeriesData(name, formattedData);
+      scalarChart.commitChanges();
+    };
+  }.call(this);
+
+  @property({type: Object})
+  getDataLoadUrl: Function = ({tag, run}) => {
+    return getRouter().pluginRoute(
+      'scalars',
+      '/scalars',
+      new URLSearchParams({tag, run})
+    );
+  };
+
+  // To be provided as the `url-fn` property to `tf-downloader`.
+  @property({type: Object})
+  _downloadUrlFn: object = (tag, run) => this.getDataLoadUrl({tag, run});
+
+  // A function called to fetch the scalars data from the backend.
+  // Should receive a {tag, run, experiment} object and return
+  // a promise resolving with the fetched scalars. The default
+  // implementation of this function executes:
+  // this.requestManager.request(
+  //      this.getDataLoadUrl({tag, run, experiment})
+  @property({type: Object})
+  requestData: object;
+
+  @property({type: Object})
+  _getDataLoadName: object = (datum) => this._getSeriesNameFromDatum(datum);
+
+  @property({
+    type: Boolean,
+    reflectToAttribute: true, // for CSS
+  })
+  _expanded: boolean = false;
+
+  @property({type: Boolean})
+  _logScaleActive: boolean;
+
+  @property({type: Array})
+  _tooltipColumns: unknown[] = function() {
+    const columns = DEFAULT_TOOLTIP_COLUMNS.slice();
+    const ind = columns.findIndex((c) => c.title == 'Name');
+    columns.splice(ind, 1, {
+      title: 'Name',
+      evaluate: (d) => {
+        const datum = d.dataset.metadata().meta;
+        return this._getSeriesDisplayNameFromDatum(datum);
+      },
+    });
+    return columns;
+  }.call(this);
```

Comment:
Since arrow functions implicitly bind `this`, we’d generally prefer to
write this as:

```typescript
_tooltipColumns = (() => {
  // ...
  return columns;
})();
```

The Google JS style guide says:

> Prefer arrow functions over the `function` keyword, particularly for
> nested functions

<https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions>

and the internal-only TypeScript style guide is stronger with “Always
use arrow functions instead of pre-ES6 function expressions”.

Reply:
Done.